modules/logging_config/logging.nix

```diff
@@ -0,0 +1,30 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+{
+  options.logging-service.enable = lib.mkOption {
+    type = lib.types.bool;
+    default = false;
+    description = "Enable or disable CAN logging service";
+  };
+
+  config = lib.mkIf config.logging-service.enable {
+    systemd.services.logging-service = {
+      description = "CAN Logging Service";
+      wantedBy = [ "multi-user.target" ];
+      before = [ "network.target" ];
+
+      serviceConfig = {
+        Type = "simple";
+        ExecStart = "${pkgs.bash}/bin/bash ${./log.sh}";
```

Comment:
i don't remember exactly how it works but make sure the shell script is succcessfully installed for this to run.